features/atcryptoauth/sample_ssl_client.cpp

```diff
@@ -0,0 +1,393 @@
+
+    // Get own certificate. Save it in the code.
+    ret += mbedtls_x509_crt_parse( &clientcert, (const unsigned char *)client_cert_pem, sizeof(client_cert_pem) );
+    // Register ATCA PK context
+    ret = mbedtls_atca_pk_setup( &clientkey, ATCA_ECC_KEY_ID_0 );
+    if( ret < 0 )
+    {
+        mbedtls_printf( " failed\r\n  !  mbedtls_atca_pk_setup returned error!\r\n\n" );
+        goto exit;
+    }
+    mbedtls_printf( " ok\r\n" );
+
+    /*
+     * 3. Start the connection
+     */
+    mbedtls_printf( "  . Connecting to tcp/%s/%d ...", SERVER_NAME, SERVER_PORT );
+
+    if( ( ret = mbedtls_net_connect( SERVER_NAME, SERVER_PORT ) ) != 0 )
+    {
+        mbedtls_printf( " failed\r\n  ! mbedtls_net_connect returned %d\r\n\n", ret );
+        goto exit;
+    }
+
+    mbedtls_printf( " ok\r\n" );
+
+
+    /*
+     * 4. Setup stuff
+     */
+    mbedtls_printf( "  . Setting up the SSL/TLS structure ..." );
+
+    if( ( ret = mbedtls_ssl_config_defaults( &conf,
+                    MBEDTLS_SSL_IS_CLIENT,
+                    MBEDTLS_SSL_TRANSPORT_STREAM,
+                    MBEDTLS_SSL_PRESET_DEFAULT ) ) != 0 )
+    {
+        mbedtls_printf( " failed\r\n  ! mbedtls_ssl_config_defaults returned %d\r\n\n", ret );
+        goto exit;
+    }
+
+    mbedtls_printf( " ok\r\n" );
+
+    /* OPTIONAL is not optimal for security,
+     * but makes interop easier in this simplified example */
+    mbedtls_ssl_conf_authmode( &conf, MBEDTLS_SSL_VERIFY_REQUIRED );
+    mbedtls_ssl_conf_ca_chain( &conf, &cacert, NULL );
+    mbedtls_ssl_conf_own_cert( &conf, &clientcert, &clientkey );
+    mbedtls_ssl_conf_rng( &conf, mbedtls_ctr_drbg_random, &ctr_drbg );
+#ifdef SSL_CLIENT_APP_ACTIVATE_DEBUG
+    mbedtls_ssl_conf_dbg( &conf, my_debug, stdout );
+#endif /* SSL_CLIENT_APP_ACTIVATE_DEBUG */
+
+    if( ( ret = mbedtls_ssl_setup( &ssl, &conf ) ) != 0 )
+    {
+        mbedtls_printf( " failed\r\n  ! mbedtls_ssl_setup returned %d\r\n\n", ret );
+        goto exit;
+    }
+
+    if( ( ret = mbedtls_ssl_set_hostname( &ssl, SERVER_NAME ) ) != 0 )
+    {
+        mbedtls_printf( " failed\r\n  ! mbedtls_ssl_set_hostname returned %d\r\n\n", ret );
+        goto exit;
+    }
+
+    mbedtls_ssl_set_bio( &ssl, (void *)_tcpsocket, mbedtls_net_send, mbedtls_net_recv, NULL );
+
+
+    /*
+     * 5. Handshake
+     */
+    mbedtls_printf( "  . Performing the SSL/TLS handshake ..." );
+
+    while( ( ret = mbedtls_ssl_handshake( &ssl ) ) != 0 )
+    {
+        if( ret != MBEDTLS_ERR_SSL_WANT_READ && ret != MBEDTLS_ERR_SSL_WANT_WRITE )
+        {
+            mbedtls_printf( " failed\r\n  ! mbedtls_ssl_handshake returned -0x%x\n\r\n", -ret );
+            goto exit;
+        }
+    }
+
+    mbedtls_printf( " ok\r\n" );
+
+
+    /*
+     * 6. Verify the server certificate
+     */
+    mbedtls_printf( "  . Verifying peer X.509 certificate ..." );
+
+    if( ( flags = mbedtls_ssl_get_verify_result( &ssl ) ) != 0 )
```

Comment:
Since you used `VERIFY_REQUIRED` above, this call is irrelevant: you already know the verification succeeded. Remove the whole “Verify the server certificate” section.